# Financial Forecasting: The Predictive Power

Financial forecasting involves using known historical relationships to estimate future financial outcomes. The "Model" we build today becomes a "Calculator" for tomorrow.

Business Scenario: Stock Price Prediction
You are a financial analyst trying to predict the Stock Price (Y) of a technology firm. You believe three macroeconomic factors drive the price:
$X_1$ Interest Rates (%) – Generally, when rates go up, stock prices go down (inverse relationship).

$X_2$: GDP Growth (%) – A stronger economy usually means higher stock prices.

$X_3$: Quarterly Earnings Growth (%) – The company's internal performance.

## Lab Practice: Building the Forecast Model

In [ ]:
import pandas as pd
import statsmodels.api as sm

# 1. Historical Financial Data (Last 12 Quarters)
data = {
    'Stock_Price': [150, 155, 160, 158, 165, 170, 175, 182, 180, 190, 195, 200],
    'Interest_Rate': [3.0, 3.1, 3.2, 3.5, 3.4, 3.2, 3.0, 2.8, 3.2, 3.0, 2.9, 2.7],
    'GDP_Growth': [2.1, 2.2, 2.1, 1.9, 2.0, 2.3, 2.5, 2.6, 2.4, 2.7, 2.8, 2.9],
    'Earnings_Growth': [5, 6, 5, 4, 7, 8, 9, 10, 8, 11, 12, 13]
}
df = pd.DataFrame(data)

# 2. Fit the Model
X = df[['Interest_Rate', 'GDP_Growth', 'Earnings_Growth']]
X = sm.add_constant(X)
Y = df['Stock_Price']
forecast_model = sm.OLS(Y, X).fit()

# 3. THE FORECASTING STEP
# Scenario: What if Interest Rates jump to 4.0%, GDP is 2.0%, and Earnings grow by 5%?
future_scenario = [1, 4.0, 2.0, 5.0] # 1 is for the constant
prediction = forecast_model.predict([future_scenario])

print(forecast_model.summary())
print(f"\nFORECASTED STOCK PRICE: ${prediction[0]:.2f}")

                            OLS Regression Results                            
Dep. Variable:            Stock_Price   R-squared:                       0.964
Model:                            OLS   Adj. R-squared:                  0.951
Method:                 Least Squares   F-statistic:                     72.43
Date:                Mon, 11 May 2026   Prob (F-statistic):           3.86e-06
Time:                        02:36:19   Log-Likelihood:                -30.037
No. Observations:                  12   AIC:                             68.07
Df Residuals:                       8   BIC:                             70.01
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
                      coef    std err          t      P>|t|      [0.025      0.975]
-----------------------------------------------------------------------------------
const             -39.1290     57.368     

**Predictive Power**

R-squared (0.964): Your model explains 96.4% of the movement in the stock price. In the world of finance, an R-squared this high is excellent, but it often suggests that your variables are very closely "tracking" the price.

F-statistic (72.43) & Prob (3.86e-06): The model is statistically significant as a whole. It is definitely better than guessing.

**The Coefficients (The "Sensitivity" Drivers)**
This is where the "What-If" analysis happens. Let's look at how the stock reacts to different economic changes:

- Interest_Rate (30.72): This is your most significant variable (p = 0.015). Surprisingly, in this specific dataset, the coefficient is positive. This means for every 1% increase in interest rates, the stock price actually rose by $30.72.
A positive coefficient here might suggest this is a "Financial Sector" stock (like a bank) that earns more money when interest rates are higher.

- GDP_Growth (39.24): This is "borderline" significant (p = 0.056). It is just barely over our 0.05 cutoff. It suggests a strong economy helps the stock, but we aren't 95% certain yet.

- Earnings_Growth (3.00): This is not significant (p = 0.096). Interestingly, the company's own internal growth seems to matter less than the external macro environment (Interest Rates and GDP).

The Forecast Interpretation
Your Forecasted Price: $177.28

This was calculated using your "Future Scenario":

Interest Rate: 4.0%

GDP Growth: 2.0%

Earnings Growth: 5.0%

Even though Interest Rates jumped to 4.0%, the model predicted a relatively high price because your "Interest Rate" coefficient is positive.

Wait! Here is the "Reality Check" for a Financial Analyst:
Look at the const (-39.12). The p-value is 0.514. This means the starting point of your model is very unstable. In finance, we call this "Model Risk." If your intercept isn't reliable, your final forecasted price ($177.28) should be treated as an estimate with a wide margin of error, not a guaranteed fact.

## Derive Regression Equation

Based on the OLS output you generated, the model built this specific math formula:

$$
\text{Stock Price} = \text{const}
+ \beta_1 \cdot \text{Interest Rate}
+ \beta_2 \cdot \text{GDP Growth}
+ \beta_3 \cdot \text{Earnings Growth}
$$

Plugging in the specific Coefficients (coef):
$$
\text{Stock Price} = -39.1290
+ 30.7277 \cdot X_1
+ 39.2458 \cdot X_2
+ 3.0012 \cdot X_3
$$



### Plugging in the Scenario

You provided the values for \(X_1\), \(X_2\), and \(X_3\).  
The Python code substituted those values into the regression equation and produced the following components:

- **Intercept:**  
  \(-39.1290\)

- **Interest Rate Impact:**  
  \(30.7277 \times 4.0 = 122.9108\)

- **GDP Impact:**  
  \(39.2458 \times 2.0 = 78.4916\)

- **Earnings Impact:**  
  \(3.0012 \times 5.0 = 15.0060\)


  

$$
-39.1290 + 122.9108 + 78.4916 + 15.0060 = 177.2794
$$

$$
\text{Predicted Stock Price} = \$177.28
$$

## Why Is This Useful for a Business?

Imagine you are the **CFO**. You don’t know for sure what the Federal Reserve will do with interest rates.  
Using the regression model, you can run multiple “what-if” scenarios to understand how different economic environments might affect your stock price.

### Example Scenarios

- **Optimistic Scenario**  
  - Interest Rate = 2.0%  
  - GDP Growth = 4.0%

- **Base Scenario** *(your current case)*  
  - Interest Rate = 4.0%  
  - GDP Growth = 2.0%

- **Pessimistic Scenario**  
  - Interest Rate = 6.0%  
  - GDP Growth = 0.5%